### PR TITLE
fix: listen function

### LINKED ISSUE #226: [Bug]: It's not possible to listen to wallet events.
### Issue description

When calling the listen function an error is thrown. Potentially, because there is a mismatch in names between `listen` and `listenRust` in wallet/bindings/nodejs/bindings.ts.

### Version

develop

### Expected behaviour

It's possible to listen to events.

### Actual behaviour

The following error is thrown in Firefly: `Error: s is not a function`

The following error is thrown when running nodejs example 7: `Error: listenRust is not a function`

### Can the issue reliably be reproduced?

Yes

### Steps to reproduce the issue

Run 7-events.js

### Errors

The following error is thrown in Firefly: `Error: s is not a function`

The following error is thrown when running nodejs example 7: `Error: listenRust is not a function`

### Duplicate declaration

- [X] I have searched the issues tracker this issue and there is none

### PR DESCRIPTION
**Pull request by Tuditi**
# Description of change

Fix the listen function

## Links to any relevant issues

closes: https://github.com/iotaledger/iota-sdk/issues/226

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

Ran FF with the linked package.

Ran example 7-events.js (still fails due to a change introduced while merging sendAmount() & sendMicroAmount())

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by thibault-martinez — CHANGES_REQUESTED** [review]
.change

**Comment by Thoralf-M** [comment]
Is the wasm binding also working with this?

**Comment by Tuditi** [comment]
> Is the wasm binding also working with this?

Not as far as I can see from this file:
https://github.com/iotaledger/iota-sdk/blob/6195e9a32edab969c0df858c507c1890b1f26320/sdk/src/wallet/bindings/wasm/lib/bindings.ts#L10

File: .changes/fix-listen.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"client-nodejs-binding": patch
```
Comment (thibault-martinez) [inline .changes/fix-listen.md]:
```suggestion
"wallet-nodejs-binding": patch
```

**Review by thibault-martinez — DISMISSED** [review]

**Comment by Thoralf-M** [comment]
@Tuditi for the wasm binding the JS lib is copied over, apart from https://github.com/iotaledger/iota-sdk/blob/6195e9a32edab969c0df858c507c1890b1f26320/sdk/src/wallet/bindings/wasm/lib/bindings.ts
and there it still exports `listen`, so `listenTo` will not be found 
So I think the way this PR currently is would just change it around, so it's found on the nodejs binding, but will fail on wasm

**Comment by Thoralf-M** [comment]
Maybe also enable the CI for wasm by adding `- "sdk/src/client/bindings/nodejs/**" # Re-include nodejs bindings, because lib is copied from there` here
https://github.com/iotaledger/iota-sdk/blob/d4c1d9a876cfc470d9ea6b290fc4f703f90178bf/.github/workflows/bindings-wasm.yml#L15-L28

`"sdk/src/wallet/bindings/nodejs/**" # Re-include nodejs bindings, because lib is copied from there` here
https://github.com/iotaledger/iota-sdk/blob/784dc659ea9f2af31ebdb28a5f8500a239242951/.github/workflows/examine-wasm-bindings.yml#L10-L24

**Review by thibault-martinez — DISMISSED** [review]

**Review by thibault-martinez — DISMISSED** [review]

File: .github/workflows/bindings-wasm.yml [hunk 2]
```diff
@@ -12,7 +12,10 @@ on:
       - "!**/tests/**" # Exclude all tests
       - "!**/bindings/**" # Exclude all bindings
       - "!cli/**" # Exclude CLI
-      - "sdk/src/client/bindings/wasm/**" # Re-include wasm bindings
+      - "sdk/src/client/bindings/wasm/**" # Re-include wasm client bindings
+      - "sdk/src/wallet/bindings/wasm/**" # Re-include wasm wallet bindings
+      - "sdk/src/client/bindings/nodejs/**" # Re-include node client bindings, b/c lib copied from there
+      - "sdk/src/wallet/bindings/nodejs/**" # Re-include node wallet bindings, b/c lib copied from there
```
Comment (Thoralf-M) [inline .github/workflows/bindings-wasm.yml]:
```suggestion
      - "sdk/src/client/bindings/wasm/**" # Re-include wasm client bindings
      - "sdk/src/client/bindings/nodejs/**" # Re-include node client bindings, b/c lib copied from there
```

File: .github/workflows/bindings-wasm.yml [hunk 3]
```diff
@@ -25,7 +28,10 @@ on:
       - "!**/tests/**" # Exclude all tests
       - "!**/bindings/**" # Exclude all bindings
       - "!cli/**" # Exclude CLI
-      - "sdk/src/client/bindings/wasm/**" # Re-include wasm bindings
+      - "sdk/src/client/bindings/wasm/**" # Re-include wasm client bindings
+      - "sdk/src/wallet/bindings/wasm/**" # Re-include wasm wasm bindings
+      - "sdk/src/client/bindings/nodejs/**" # Re-include node client bindings, b/c lib copied from there
+      - "sdk/src/wallet/bindings/nodejs/**" # Re-include node wallet bindings, b/c lib copied from there
```
Comment (Thoralf-M) [inline .github/workflows/bindings-wasm.yml]:
```suggestion
      - "sdk/src/client/bindings/wasm/**" # Re-include wasm client bindings
      - "sdk/src/client/bindings/nodejs/**" # Re-include node client bindings, b/c lib copied from there
```

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**